bundle/edu.gemini.ags/src/main/scala/edu/gemini/ags/impl/MultiProbeStrategy.scala

```diff
@@ -0,0 +1,70 @@
+package edu.gemini.ags.impl
+
+import edu.gemini.ags.api.AgsMagnitude.MagnitudeTable
+import edu.gemini.ags.api.AgsStrategy.Estimate
+import edu.gemini.ags.api.{AgsAnalysis, AgsMagnitude, AgsStrategy}
+import edu.gemini.catalog.api.CatalogQuery
+import edu.gemini.spModel.ags.AgsStrategyKey
+import edu.gemini.spModel.core.{SingleBand, RBandsList, SiderealTarget}
+import edu.gemini.spModel.guide.{ValidatableGuideProbe, GuideProbe}
+import edu.gemini.spModel.obs.context.ObsContext
+
+import scala.concurrent.Future
+
+import scala.concurrent.ExecutionContext.Implicits.global
+
+/**
+ * Represents an AGS strategy consisting of multiple substrategies.
+ *
+ * There are several inherent shortcomings with this:
+ * 1. select requires a position angle, and querying each substrategy may return a number of position angles.
+ *    In the current implementation, the position angle for the first strategy is returned.
+ * 2. probeBands must return one of a fixed number of BandsLists. We currently try to find a common one, or one
+ *    that is a subset of all of those of the substrategies. If that is not possible, we try to find a single band
+ *    that is a subset of those of the substrategies. If that fails, we just return R.
+ */
+case class MultiProbeStrategy(key: AgsStrategyKey, strategies: List[AgsStrategy]) extends AgsStrategy {
```

Comment:
I fear this class :smile: I think it should be thoroughly tested like `ScienceProbeStrategy`

Some things that worry me:
* What does it mean to do an analyze? Is it per probe?
* catalog queries support multiple queries but it is just to accomodate GeMS and I was planning to remove it as you can do GeMS with a single query
* I think `probeBands` should be a superset of bands rather than a subset, but I'm ready to be convinced otherwise